packages/expo-build-properties/src/android.ts

```diff
@@ -261,3 +261,31 @@ export const withAndroidQueries: ConfigPlugin<PluginConfigType> = (config, props
     return config;
   });
 };
+
+/**
+ * This plugin is used to set the `kotlinVersion` in the `build.gradle` file for the
+ * kotlin gradle plugin.
+ */
+export const withAndroidKotlinGradlePluginVersion: ConfigPlugin<PluginConfigType> = (
+  config,
+  props
+) => {
+  if (props.android?.kotlinVersion == null) {
+    return config;
+  }
+  return withDangerousMod(config, [
+    'android',
+    async (config) => {
```

Comment:
```suggestion
  return withProjectBuildGradle(config, async (config) => {
```

can use `withProjectBuildGradle` directly